src/wallet/utils.rs

```diff
@@ -97,6 +101,31 @@ impl<Pk: MiniscriptKey> Satisfier<Pk> for Older {
     }
 }
 
+pub(crate) trait ToDerivationCtx {
+    fn to_derivation_ctx<'secp>(
+        self,
+        secp_ctx: &'secp Secp256k1<All>,
+    ) -> DescriptorPublicKeyCtx<'secp, All>;
+}
+
+impl ToDerivationCtx for bip32::ChildNumber {
+    fn to_derivation_ctx<'secp>(
+        self,
+        secp_ctx: &'secp Secp256k1<All>,
+    ) -> DescriptorPublicKeyCtx<'secp, All> {
+        DescriptorPublicKeyCtx::new(secp_ctx, self)
+    }
+}
+
+impl ToDerivationCtx for () {
+    fn to_derivation_ctx<'secp>(
+        self,
+        secp_ctx: &'secp Secp256k1<All>,
+    ) -> DescriptorPublicKeyCtx<'secp, All> {
+        DescriptorPublicKeyCtx::new(secp_ctx, bip32::ChildNumber::Normal { index: 0 })
+    }
+}
```

Comment:
What's the benefit to doing this? When I first read `().to_derivation_ctx()` I was very confused about what was happening and figured it was some obscure syntax I didn't know.

Might be preferable to just have a `make_derivation_ctx` function since `to_derivation_ctx(...)` is more straightforward to understand than `().to_derivation_ctx(...)`.

But I don't have a strong opinion here ...

Reply:
Yeah that was a leftover from the initial implementation where I was creating both contexts, some of them with a specific child_number and some others with a dummy one. Now I think I'm only using the dummy context, so I'll just remove that part entirely